pkg/kubelet/pleg/generic.go

```diff
@@ -475,7 +473,7 @@ func (g *GenericPLEG) updateCache(ctx context.Context, pod *kubecontainer.Pod, p
 	// Evented PLEG after the event has been received by the Kubelet.
 	// For more details refer to:
 	// https://github.com/kubernetes/enhancements/tree/master/keps/sig-node/3386-kubelet-evented-pleg#timestamp-of-the-pod-status
-	if utilfeature.DefaultFeatureGate.Enabled(features.EventedPLEG) && isEventedPLEGInUse() && status != nil {
```

Comment:
I see a lot of removals of the faeture gate check , are we keeping this feature well guarded? we regressed for several releases and had to downgrade it recently, so please be sure this code is well guarded

Reply:
This is from https://github.com/kubernetes/kubernetes/pull/122778/files#diff-b010d1c9ff51c8975875be97aeebd3c8c6206128f520115752f9dc007e8afcd1L106.

Reply:
yeah, but my question is if the code is feature gated, non of this code should run if the feature gate is not explicitly enabled